bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -526,6 +548,32 @@ object BagsManager {
           }
         }
       }
+
+
+    def applySelection(ctx: TpeContext): Unit = {
+      val oldEnv = ctx.targets.envOrDefault
+
+      // If AGS results were found, apply them to the target env; otherwise, clear out any existing auto group.
+      val newEnv = selOpt.map(_.applyTo(oldEnv)).getOrElse {
+        val oldGuideEnv = oldEnv.getGuideEnvironment.guideEnv
+        if (oldGuideEnv.auto === AutomaticGroup.Initial || oldGuideEnv.auto === AutomaticGroup.Disabled) oldEnv
+        else oldEnv.setGuideEnvironment(GuideEnvironment(GuideEnv(AutomaticGroup.Initial, oldGuideEnv.manual)))
+      }
+
+      // Commit all the changes inside a write lock, but first make sure that
+      // the underlying observation has not changed from what we expected.  If
+      // it has changed, then a new BAGS calculation will be triggered anyway.
+
+      ctx.obsShell.foreach { obs =>
+        obs.getProgramWriteLock()
+        try {
+          if (expectedHash.forall(hashObs(obs).contains)) {
```

Comment:
Only apply the results if a newly computed AGS hash matches what it was before (or if we don't supply an expected hash at all).

Reply:
When the hash no longer matches, it means that the observation has been updated in some way that will impact the AGS calculation.  In turn that means that a new AGS calculation will be kicked off anyway.